### ISSUE TITLE
[BUG] Pylint >=2.13 crashes when using parallel execution.

### ISSUE THREAD

**Issue by koniiiik** [PyCQA/prospector#513]
**Describe the bug**
Pylint versions above 2.13 are incompatible with parallel execution through prospector; they lead to an IO error in the dill package used to serialize results.

**To Reproduce**
1. install prospector 1.7.7, pylint 2.13 or 2.14
2. enable parallel pylint execution in .prospector.yml with
```
pylint:
  options:
    jobs: 0
```
4. Call prospector with `prospector -X`
5. The following stack trace
```
#  prospector -X
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/prospector/run.py", line 94, in execute
    messages += tool.run(found_files)
  File "/usr/local/lib/python3.8/site-packages/prospector/tools/pylint/__init__.py", line 253, in run
    self._linter.check(self._args)
  File "/usr/local/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 650, in check
    check_parallel(
  File "/usr/local/lib/python3.8/site-packages/pylint/lint/parallel.py", line 141, in check_parallel
    jobs, initializer=initializer, initargs=[dill.dumps(linter)]
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 364, in dumps
    dump(obj, file, protocol, byref, fmode, recurse, **kwds)#, strictio)
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 336, in dump
    Pickler(file, protocol, **_kwds).dump(obj)
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 620, in dump
    StockPickler.dump(self, obj)
  File "/usr/local/lib/python3.8/pickle.py", line 485, in dump
    self.save(obj)
  File "/usr/local/lib/python3.8/pickle.py", line 601, in save
    self.save_reduce(obj=obj, *rv)
  File "/usr/local/lib/python3.8/pickle.py", line 715, in save_reduce
    save(state)
  File "/usr/local/lib/python3.8/pickle.py", line 558, in save
    f(self, obj)  # Call unbound method with explicit self
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 1251, in save_module_dict
    StockPickler.save_dict(pickler, obj)
  File "/usr/local/lib/python3.8/pickle.py", line 969, in save_dict
    self._batch_setitems(obj.items())
  File "/usr/local/lib/python3.8/pickle.py", line 995, in _batch_setitems
    save(v)
  File "/usr/local/lib/python3.8/pickle.py", line 601, in save
    self.save_reduce(obj=obj, *rv)
  File "/usr/local/lib/python3.8/pickle.py", line 715, in save_reduce
    save(state)
  File "/usr/local/lib/python3.8/pickle.py", line 558, in save
    f(self, obj)  # Call unbound method with explicit self
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 1251, in save_module_dict
    StockPickler.save_dict(pickler, obj)
  File "/usr/local/lib/python3.8/pickle.py", line 969, in save_dict
    self._batch_setitems(obj.items())
  File "/usr/local/lib/python3.8/pickle.py", line 995, in _batch_setitems
    save(v)
  File "/usr/local/lib/python3.8/pickle.py", line 558, in save
    f(self, obj)  # Call unbound method with explicit self
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 1405, in save_file
    f = _save_file(pickler, obj, open)
  File "/usr/local/lib/python3.8/site-packages/dill/_dill.py", line 1379, in _save_file
    position = obj.tell()
io.UnsupportedOperation: underlying stream is not seekable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/prospector", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/prospector/run.py", line 202, in main
    prospector.execute()
  File "/usr/local/lib/python3.8/site-packages/prospector/run.py", line 112, in execute
    raise FatalProspectorException from ex
TypeError: __init__() missing 1 required positional argument: 'message'
```

**Environment (please complete the following information):**
 - OS: The official python:3.8 docker image
 - Tool: pylint
 - Prospector version: 1.7.7
- Python version: 3.8.6

**Additional context**
Downgrading pylint to a version prior to 2.13 makes prospector run correctly with these settings.

**Comment by smac89** [comment]
Experiencing the exact same thing

**Comment by koniiiik** [comment]
Adding some more details that I managed to find out by poking around the code. Apparently, the object that fails to get pickled by dill is the `out` attribute of `prospector.tools.pylint.collector.Collector`, which gets set to a `TextIOWrapper` for stdout (since `Collector.__init__` sets the `output` argument of `BaseReporter` to `None`, which defaults to `sys.stdout`), but then when pylint is actually executed, `CaptureOutput` fiddles with `sys.stdout` and `sys.stderr`, which confuses dill.

I've attempted a simple fix in #526 which makes prospector run successfully with the latest pylint, but I'm not familiar enough with the internals of pylint and prospector to be confident that this is the correct solution. Any additional eyeballs would be appreciated.
